src/main/java/org/springframework/data/elasticsearch/repository/support/querybyexample/ExampleCriteriaMapper.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.repository.support.querybyexample;
+
+import org.springframework.dao.InvalidDataAccessApiUsageException;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.ExampleMatcher;
+import org.springframework.data.elasticsearch.core.mapping.ElasticsearchPersistentEntity;
+import org.springframework.data.elasticsearch.core.mapping.ElasticsearchPersistentProperty;
+import org.springframework.data.elasticsearch.core.query.Criteria;
+import org.springframework.data.mapping.PersistentPropertyAccessor;
+import org.springframework.data.mapping.context.MappingContext;
+import org.springframework.data.support.ExampleMatcherAccessor;
+import org.springframework.lang.Nullable;
+import org.springframework.util.StringUtils;
+
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Maps a {@link Example} to a {@link org.springframework.data.elasticsearch.core.query.Criteria}
+ *
+ * @param <T> Class type
+ * @author Ezequiel Antúnez Camacho
+ */
+class ExampleCriteriaMapper<T> {
+
+	private static final Set<ExampleMatcher.StringMatcher> SUPPORTED_MATCHERS = EnumSet.of(
+			ExampleMatcher.StringMatcher.DEFAULT, ExampleMatcher.StringMatcher.EXACT, ExampleMatcher.StringMatcher.STARTING,
+			ExampleMatcher.StringMatcher.CONTAINING, ExampleMatcher.StringMatcher.ENDING);
+
+	private final MappingContext<? extends ElasticsearchPersistentEntity<T>, ElasticsearchPersistentProperty> mappingContext;
```

Comment:
I'm using the mapping context provided by ElasticsearchOperations and ReactiveElasticsearchOperations, which returns MappingContext<? extends ElasticsearchPersistentEntity<?>, ElasticsearchPersistentProperty> not SimpleElasticsearchMappingContext. 
But internally the implementation provided by operation clases is in fact SimpleElasticsearchMappingContext.
If ExampleCriteriaMapper depends on the interface, instead on the specific implementation, we don't need to instantiate it directly or cast it from the operations to the implementation. Also, if for any reason operation classes replace the implementation of the MappingContext the ExampleCriteriaMapper won't be affected.
Anyway I have removed the parametrizable definition of ExampleCriteriaMapper, since at this point is not required. I'll update my changes soon.

Reply:
Updated in https://github.com/spring-projects/spring-data-elasticsearch/pull/2422/commits/0b36a268b21075a71a4a3447c864fa3b968e6356